trade-finance-manager-api/src/graphql/reducers/mappings/facilities/mapCoverEndDate.js

```diff
@@ -3,6 +3,10 @@ const { hasValue } = require('../../../../utils/string');
 const { formatYear } = require('../../../../utils/date');
 
 const mapCoverEndDate = (day, month, year) => {
+  // checks if valid date - not valid if coverEndDate is not set
+  if (Number.isNaN(Number(day)) || Number.isNaN(Number(month)) || Number.isNaN(Number(year))) {
+    return undefined;
+  }
```

Comment:
I think this should be combined into the `hasCoverEndDate` function and should use `Number.isInteger()` instead of `isNaN`.

In fact this approach makes it confusing to me - the `hasValue` checks below check if it's a string. But here we're making sure it's an integer? And then when the `coverEndDate`is set, it transforms day/month/year into a number

What are the expected inputs here for GEF and BSS? Maybe one is string, one is number?

Reply:
done